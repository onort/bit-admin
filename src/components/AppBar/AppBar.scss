@import "../../styles/main.scss";

.container {
  z-index: 11;
  padding: $spacer-sm;
  height: $appBar-height;
  box-shadow: 0px 1px 10px #999;
}

.appBar {
  display: flex;
  align-items: stretch;
  align-self: center;
}

.menuIcon {
  display: flex;
  align-items: center;
  font-size: $font-size-xlg;
  margin-right: $spacer-xlg;
  cursor: pointer;
  transition: color 0.5s;

  &:hover {
    color: lighten($text-color, 15);
  }
}

.brandContainer {
  display: flex;
  margin: 0 $spacer-sm;
}

.brand {
  font-size: $brand-font-size;
  font-weight: 600;
  color: lighten($text-color, 10);
}
