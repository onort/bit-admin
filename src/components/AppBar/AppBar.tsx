import React from "react"

import styles from "./AppBar.scss"

const AppBar: React.SFC = props => {
  return (
    <header className={styles.container}>
      <div className={styles.appBar}>
        <div className={styles.brandContainer}>
          <span className={styles.brand}>Bmat Bit Admin</span>
        </div>
      </div>
    </header>
  )
}

export default AppBar
