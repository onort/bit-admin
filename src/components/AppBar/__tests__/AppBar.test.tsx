import React from "react"
import { shallow } from "enzyme"

import AppBar from "../"

describe("<AppBar />", () => {
  it("should match snapshot", () => {
    const toggleSidebar = jest.fn()
    const wrapper = shallow(<AppBar toggleSidebar={toggleSidebar} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should fire toggleSidebar when menu icon is clicked", () => {
    const toggleSidebar = jest.fn()
    const wrapper = shallow(<AppBar toggleSidebar={toggleSidebar} />)
    wrapper.find(".menuIcon").simulate("click")
    expect(toggleSidebar).toBeCalledTimes(1)
  })
})
