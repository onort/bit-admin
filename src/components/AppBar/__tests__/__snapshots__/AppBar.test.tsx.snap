// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should match snapshot 1`] = `
<header
  className="container"
>
  <div
    className="appBar"
  >
    <div
      className="brandContainer"
    >
      <span
        className="menuIcon"
        onClick={[MockFunction]}
      >
        <MdMenu />
      </span>
      <span
        className="brand"
      >
        Bmat Bit Admin
      </span>
    </div>
  </div>
</header>
`;
